### DISCUSSION TITLE
My Filters is not a trusted source

### DISCUSSION THREAD

**Discussion by stonedcarrot** [uBlockOrigin/uAssets#21038] [Q&A] [answered]
Hello, I'm new to this I usually just use ubo as a non advanced user so likely the problem is between the chair and the keyboard.
anyway I'm having a weird issue where a filter such as: "example.*##+js(trusted-set, targer, value)" and the trusted-set becomes highlighted in red and filter fails to work. 
![image](https://github.com/uBlockOrigin/uAssets/assets/152811986/083e7d21-8532-42f7-acf5-f789cca2302b)
If I could have some help that would be great!

**Comment by ghajini** [comment]
read release notes https://github.com/gorhill/uBlock/releases/tag/1.53.0
https://github.com/gorhill/uBlock/commit/64c1f8767c

**Comment by ghost** [comment, marked as answer]
❗ Be careful, do not trust any random filter, not even your own if you're not sure about what you are doing
❗ Also, make sure not to tamper with any other settings on accident.

To allow uBlock Origin to trust `My filters`, you can:

1. Go to uBlock Origin's settings
2. Make sure `I am an advanced user` is ticked / checked.
3. Click on the gear (⚙️) icon near `I am an advanced user`
4. At `trustedListPrefixes` add `user-` so that you have `ublock- user-` (assuming you're not trusting any other maintainer)

**Reply by gorhill** [reply]
In addition to the above instructions, _My filters_ are trusted by default in dev build.

**Reply by ghost** [reply]
@gorhill Regarding trust / security, is there any measure to prevent damage in case any of the CDN's starts serving hidden malware?

**Reply by gorhill** [reply]
No. If you do not trust CDNs, you can wholly disable filters requiring a trusted source by setting `trustedListPrefixes` to `-`.

**Reply by ghost** [reply]
Well yeah, but most people don't know how ad blockers work. Those people could use online banking, send sensitive information etc.

**Reply by ghost** [reply]
Isn't this similar to the vulnerability described by different media outlets a couple years ago?

**Reply by gorhill** [reply]
> described by different media outlets a couple years ago?

I can't give answer to vague question. If you mean ABP's `rewrite`, no it's not because there was no concept of trusted lists and that filter option could be used in any list.

In any case, as said, if you don't trust CDNs, remove all trusted sources, if you don't trust uBO, don't use it -- this is the advice I give to everybody who raise trust issues.

**Reply by stonedcarrot** [reply]
Sweet, this worked a charm! Thank you very much :)

**Comment by Dean-Corso** [comment]
Hey guys,

so I found that trusted option to enable it in uBO to make those specific trusted filter rules working. I would also like to ask what the disadvantage is to enable it (user-)? Do you have some example what could be dangerous etc? Just would like to understand it whether it could be really a problem to have this prefix enabled.

So what I can see is that different filter lists I can enable in uBO using the trusted prefix but in case of my own filters I have to enable it extra. Why? I have enabled all own uBO filter scripts & uBO Cookie Notices which is using many trusted rules. I don't use any extern filter lists. Could that bring any problems in that case when just using official uBO scripts & own filters?

If I understand it right then the user- does allow every filter to use the trusted rules including own filters. So what about to separate an option to allow all trusted prefixes just in own custom filters without to allow it for all other filter lists (except the uBO lists)? Just a example, so in my case I have few websites I would like to set some trusted cookies / local etc and I need to enable user- in uBO settings to make them work. Just don't know whether it makes sense to allow the trusted prefix for ALL when I just wanna enable them for my OWN custom filters you know. So do you guys have some advice for that situation or some tips how to deal with that etc? Don't do anything wrong. Thanks.

**Reply by garry-ut99** [reply]
You're asking questions which are already accurately answered, like you ignored the already existing issues, discussions, comments and documentation :
- https://github.com/uBlockOrigin/uAssets/discussions/18589
- https://github.com/uBlockOrigin/uBlock-issues/discussions/2895
- https://github.com/gorhill/uBlock/commit/64c1f8767c 

In the aforementioned links is already clearly stated that what is dangerous is when novice users paste to uBO random risky filters from internet, or create own risky filters, which can be malicious, unlike filters from default uBO lists, which are safe because are created by trusted and experienced developers, therefore filters from uBO default lists aren't so much risky and don't need to be disabled.

> Dean-Corso : If I understand it right then the `user-` does (...)

https://github.com/gorhill/uBlock/commit/64c1f8767c

**Comment by stephenhawk8054** [comment]
@gorhill I think `badfilter` doesn't work with trusted filter (for example `uritransform`) if `My filters` is not marked as trusted source?

Tested filter: `||rambler.ru/*#rcmrclid=$uritransform=/#rcmrclid=.*//` in uBO privacy list

![image](https://github.com/uBlockOrigin/uAssets/assets/66517106/359fb5e4-297a-4ee5-8a72-da084ff95004)

**Reply by gorhill** [reply]
https://github.com/uBlockOrigin/uBlock-issues/issues/3136
